sklearn/ensemble/tests/test_iforest.py

```diff
@@ -322,37 +325,3 @@ def test_iforest_deprecation():
     warn_msg = "'behaviour' is deprecated in 0.22 and will be removed in 0.24"
     with pytest.warns(FutureWarning, match=warn_msg):
         iforest.fit(iris.data)
-
-
-def test_iforest_with_uniform_data():
```

Comment:
We definitely don't want to remove this one.

Reply:
I don't insist on removing this, but #7141 should be resolved in a proper way then.

Reply:
thanks for opening this PR, can you develop on why #7141 was not resolved in a proper way? 
tbh I can't find what/where this issue was fixed, but this test seems to show it was fixed

Reply:
sorry just saw your previous comment - so you're saying it was fixed because the approximation error flipped side basically. I think we should go with @NicolasHug suggestion then - `is_inlier[self.decision_function(X) < -1e-15] = -1` instead of `is_inlier[self.decision_function(X) < 0] = -1`

Reply:
@ngoix Do you expect me to do it in this PR or in the separate PR?

Reply:
another PR would be the proper way, but then we'd have to merge this one only afterwards because we definitely shouldn't remove this test. 
I'd be ok to do it in this PR if it is just this one-line change